sklearn/cross_decomposition/_pls.py

```diff
@@ -45,8 +45,8 @@ def _get_first_singular_vectors_power_method(X, Y, mode="A", max_iter=500,
         # As a result, and as detailed in the Wegelin's review, CCA (i.e. mode
         # B) will be unstable if n_features > n_samples or n_targets >
         # n_samples
-        X_pinv = pinv2(X, check_finite=False)
-        Y_pinv = pinv2(Y, check_finite=False)
+        X_pinv = pinv2(X, check_finite=False, cond=10*eps)
+        Y_pinv = pinv2(Y, check_finite=False, cond=10*eps)
```

Comment:
Setting this condition forces singular values smaller than this to zero.

Reply:
I'm fine with that but if we can achieve the same stability with the estimators parameters (max_iter or tol), I think it would be preferable.

Reply:
I was able to find a setting that worked.

Reply:
Is this missing a "not"?

Reply:
yes I was missing "not"